btree/btree.c

```diff
@@ -1476,34 +1998,387 @@ static void generic_move(struct nd *src, struct nd *tgt,
 		if (nr == 0 || (nr == NR_EVEN &&
```

Comment:
one more base case condition needs to be added i.e. srcidx != -1, in case nr==NR_MAX,  srcidx may get -1 when we move all the items from source to target

Reply:
Yes but there are some cases, where we might want to move al records from src to tgt, e.g. in case of deletion, for merging , we might want to transfer all records to tgt node  and delete src node , Also, in case of insertion, whenever overflow will occur at root node, as we do not want to change pointer to root node, we might have to transfer all the data from root to some other node and fill the root with new records.

Reply:
at line 1997, 1998:
we need to set srcidx, tgtidx according to left / internal node, current implementation will work for leaf node, but for internal node it will not work, For example, if dir ==D_RIGHT and src=[(K0,c0),....(Kn, Cn),(' ',Cn+1)], according to current implementation, it will start moving record from nth index instead of n+1 th index, which may lead to inconsistent state.

Reply:
Done